[Zico] iP

zicotjia opened the pull request:
# Duke
> "Your mind is for having ideas, not holding them." - [David Allen](https://dansilvestre.com/productivity-quotes).

Duke takes away the hassle of remembering. Duke is made for goodness. Use Duke to remember stuffs. Its's:
- Text-based
- Simple
- ~~Fast~~ Super-Fast

Installation is simple,

1. Download it from [here](https://github.com/zicotjia/ip/releases/download/v0.1/duke.jar).
2. Run it.
3. Add tasks.
4. Let it manage.

This whole app is FREE!

Features:
- [x] Manage Tasks.
- [ ] Manage Deadlines.
- [ ] Reminders (coming soon).

You can customize it too with some experience in Java. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        new Duke("data/TaskList.txt").run();
    }
}
```

Commit:
Add Gradle support

zicotjia committed:
First Commit (Version 0.1)

zicotjia committed:
Fix Formatting

zicotjia committed:
Implemented Exceptions Handling v0.5

zicotjia committed:
Implemented ArrayList to hold task v0.4b

zicotjia committed:
Implemented Delete Function v0.5

zicotjia committed:
Add partial functional save & load v0.6

zicotjia committed:
Implement Storage Save and Load v0.6

zicotjia committed:
implement date data structure v0.8

zicotjia committed:
Merge branch 'branch-level-7'

zicotjia committed:
Merge branches

zicotjia committed:
change parsing logic v0.8b

zicotjia committed:
Merge branch 'add-gradle-support'

zicotjia committed:
create tasklist class

zicotjia committed:
Create Storage object

zicotjia committed:
implement Ui

zicotjia committed:
implement OOP

zicotjia committed:
add junit

zicotjia committed:
Add JavaDoc

zicotjia committed:
Make code better follow convention

zicotjia committed:
Implement find

zicotjia committed:
Merge branch 'branch-Level-9'

zicotjia committed:
merge with JavaDoc

zicotjia committed:
Merge with CodingStandard

zicotjia committed:
Improve parsing

zicotjia committed:
Improve Parsing Logic

zicotjia commented:
# Duke
> "Your mind is for having ideas, not holding them." - [David Allen](https://dansilvestre.com/productivity-quotes).

Duke takes away the hassle of remembering. Duke is made for goodness. Use Duke to remember stuffs. Its's:
- Text-based
- Simple
- ~~Fast~~ Super-Fast

Installation is simple,

1. Download it from [here](https://github.com/zicotjia/ip/releases/download/v0.1/duke.jar).
2. Run it.
3. Add tasks.
4. Let it manage.

This whole app is FREE!

Features:
- [x] Manage Tasks.
- [ ] Manage Deadlines.
- [ ] Reminders (coming soon).

You can customize it too with some experience in Java. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        new Duke("data/TaskList.txt").run();
    }
}
```

zicotjia committed:
Finished checkstyle

zicotjia committed:
Added GUI with a few bugs

gerardstevan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+        }
+    }
+
+//    /**
```
Perhaps you can delete unused commented out code?

gerardstevan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+
+import java.io.IOException;
+
+//import javafx.application.Application;
```
Maybe you should delete this commented out code?

gerardstevan reviewed (commented):
Overall, I like how your code doesn't have the common style mistake.

jchilling commented on src/main/java/duke/exceptions/DukeArrayOutOfBoundException.java:
```diff
@@ -0,0 +1,8 @@
+package duke.exceptions;
+
+public class DukeArrayOutOfBoundException extends DukeException{
+
+    public DukeArrayOutOfBoundException() {
+        super("Index cannot be zero or negative");
+    }
+}
```
I like how you created more exception classes for `DukeException`, makes exceptions a lot clearer in the code!

jchilling commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+//
+//    }
+
+//    /**
+//     * Start duke.
+//     *
+//     * @param args
+//     */
+//    public static void main(String[] args) {
+//
+//        Duke duke = new Duke("data/tasks.txt");
+//
+//        duke.run();
+//
+//    }
```
Maybe consider deleting commented code to make the whole code more concise?

jchilling commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+    public String getResponse(String line) {
+        try {
+            Command command = Parser.parse(line);
+            String res = command.execute(this.storage);
```
Perhaps name the variable as description or command to make the use of it more clear?

jchilling commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,82 @@
+
+
+    /**
+     * Changes state of task to done/
```
JavaDoc typo

jchilling reviewed (commented):
Overall I think you did a great job! There are barely any coding standard violations or coding quality issues 👍

zicotjia committed:
remove old comment

zicotjia committed:
Add assertions

There are some assertions that could be made in
Parser and Storage

Add some assertion to check existence of file
and command

zicotjia committed:
Merge pull request #1 from zicotjia/branch-A-Assertions

Create Assertions

zicotjia committed:
Improves on code quality

Some code are not following coding convention

Fix typo and rename certain variable

zicotjia committed:
Merge pull request #2 from zicotjia/branch-A-CodeQuality

Improves on code quality

zicotjia committed:
Implement duplicate handling

zicotjia committed:
Remove magic words & unused code

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Fix a bug causing delete to not working for index 1

zicotjia committed:
Merge branch 'master' of github.com:zicotjia/ip

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
fix find format

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Update README.md

zicotjia committed:
Fix bugs with linux and mac version

zicotjia committed:
Merge branch 'master' of github.com:zicotjia/ip

zicotjia committed:
Disable resizing

zicotjia committed:
Update README.md

zicotjia committed:
Modify find to search for keyword in any part of the task description

zicotjia committed:
Merge branch 'master' of github.com:zicotjia/ip

zicotjia committed:
Update README.md
